### PR TITLE
Add HTTPS support to Monitor API 

### LINKED ISSUE #2243: Add HTTPS support to Monitor API
### Problem

The Node.js-based monitor API does not support TLS/HTTPS, so we are unable to use it to monitor mainnet-public.

### Solution

- Remove `servers.ip` and `servers.port`
- Add `servers.baseUrl` with default of `http://127.0.0.1`
- Update dashboard to display newer format
- Test with `https://mainnet-public.mirrornode.hedera.com`

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR adds https support to the js monitor API

- Replace `servers[].ip` and `servers[].port` with `servers[].baseUrl` in config
- Update dashboard to use the new format
- Add configurable retry on 429 HTTP response code

**Related issue(s)**:

Fixes #2243 

**Notes for reviewer**:

This is the log with retry when running w/ mainnet-public:

```
2021-07-15T13:31:32.676-0500 INFO Running tests
2021-07-15T13:31:32.755-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/transactions?order=asc&limit=10, response status: 429, retry in 50ms
2021-07-15T13:31:32.756-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/balances?limit=10, response status: 429, retry in 50ms
2021-07-15T13:31:32.758-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/accounts?limit=10, response status: 429, retry in 50ms
2021-07-15T13:31:32.759-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/transactions?limit=1&order=desc, response status: 429, retry in 50ms
2021-07-15T13:31:32.759-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/transactions?limit=1, response status: 429, retry in 50ms
2021-07-15T13:31:32.760-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/accounts?account.balance=gte%3A0&limit=1, response status: 429, retry in 50ms
2021-07-15T13:31:32.760-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/balances?limit=10, response status: 429, retry in 50ms
2021-07-15T13:31:32.761-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/transactions?limit=1, response status: 429, retry in 50ms
2021-07-15T13:31:32.761-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/balances?limit=1&order=desc, response status: 429, retry in 50ms
2021-07-15T13:31:32.762-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/transactions?limit=10&type=credit, response status: 429, retry in 50ms
2021-07-15T13:31:32.815-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/accounts/0.0.9, response status: 429, retry in 50ms
2021-07-15T13:31:32.864-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/accounts?account.balance=gte%3A0&limit=1, response status: 429, retry in 50ms
2021-07-15T13:31:32.864-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/balances?limit=10, response status: 429, retry in 50ms
2021-07-15T13:31:32.865-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/transactions?order=asc&limit=10, response status: 429, retry in 50ms
2021-07-15T13:31:32.866-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/transactions?limit=10&type=credit, response status: 429, retry in 50ms
2021-07-15T13:31:32.867-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/transactions?limit=1, response status: 429, retry in 50ms
2021-07-15T13:31:32.868-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/balances?limit=1&order=desc, response status: 429, retry in 50ms
2021-07-15T13:31:32.869-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/balances?limit=10, response status: 429, retry in 50ms
2021-07-15T13:31:32.924-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/accounts?account.id=0.0.9&type=credit&limit=1, response status: 429, retry in 50ms
2021-07-15T13:31:32.965-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/transactions?order=asc&limit=10, response status: 429, retry in 55ms
2021-07-15T13:31:32.968-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/transactions?limit=10&type=credit, response status: 429, retry in 52ms
2021-07-15T13:31:32.969-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/transactions?limit=1, response status: 429, retry in 50ms
2021-07-15T13:31:32.979-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/balances?limit=1&order=desc, response status: 429, retry in 50ms
2021-07-15T13:31:32.985-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/balances?limit=10, response status: 429, retry in 50ms
2021-07-15T13:31:33.078-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/transactions?limit=10&type=credit, response status: 429, retry in 50ms
2021-07-15T13:31:33.081-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/balances?limit=1&order=desc, response status: 429, retry in 50ms
2021-07-15T13:31:33.083-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/balances?limit=10, response status: 429, retry in 50ms
2021-07-15T13:31:33.243-0500 WARN url: https://mainnet-public.mirrornode.hedera.com/api/v1/balances?account.id=0.0.319008, response status: 429, retry in 50ms
2021-07-15T13:31:33.349-0500 INFO Completed tests run for mainnet-public with 14/14 tests passed
```

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/monitoring/monitor_apis/config/default.serverlist.json [hunk 1]
```diff
@@ -2,12 +2,15 @@
...
     {
-      "ip": "127.0.0.1",
-      "port": 5551,
+      "baseUrl": "http://127.0.0.1:5551",
       "name": "local"
     }
   ],
   "interval": 60,
+  "retry": {
+    "max": 10,
```
Comment (steven-sheehy) [inline hedera-mirror-rest/monitoring/monitor_apis/config/default.serverlist.json:11]: Would prefer `maxAttempts` and `minBackoff` like we have these named in other components.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Looks good, one request to make testing and management easier

File: hedera-mirror-rest/monitoring/monitor_apis/config/default.serverlist.json (hunk 1 above)
Comment (xin-hedera) [reply hedera-mirror-rest/monitoring/monitor_apis/config/default.serverlist.json:11]: sure

File: hedera-mirror-rest/monitoring/monitor_apis/utils.js [hunk 2]
```diff
@@ -58,6 +60,34 @@ const getUrl = (server, path, query = undefined) => {
...
+ */
+const fetchWithRetry = async (url, opts = {}) => {
+  for (let i = 0; ; i++) {
+    const response = await fetch(url, opts);
+
+    if (response.status !== 429 || i === config.retry.max) {
+      return response;
+    }
+
+    const secondsToWait = Number(response.headers.get('retry-after'));
```
[the remark this answers is not in the document]
Comment (xin-hedera) [reply hedera-mirror-rest/monitoring/monitor_apis/utils.js:78]: will do that

**Review by steven-sheehy — APPROVED** [review]

**Review by Nana-EC — APPROVED** [review]
LGTM.
Notably, in my request for the `getBackOff` I was expecting a test file to verify varied inputs.
I realize now we don't have that on the `monitor_apis` side.
Adding one would be nice but with the push to move to java monitor I'm okay not putting much more effort since you've already verified operation E2E

**Review by ijungmann — APPROVED** [review]

**Pull request merged**